### PR TITLE
Store artifacts from CircleCI

### LINKED ISSUE #1478: Provide artifacts for Linux / Mac OSX builds
Like the Windows ones we already have.

### PR DESCRIPTION
**Pull request by heinrich5991**
Fixes #1478.

### CODE REVIEW

**Comment by heinrich5991** [comment]
@Learath2 Do you know where I can find the artifacts on CircleCI?

**Comment by def-** [comment]
You could start a CircleCI run with SSH access and try to find them there?

**Comment by Learath2** [comment]
There is supposed to be a tab on the job page. I think you need to have write access to the repo to be able to see the artifacts tab. (I can't see it on yours)

File: .circleci/config.yml [hunk 1]
```diff
@@ -52,12 +52,17 @@ jobs:
           cd build
           env CFLAGS="-Wdeclaration-after-statement -Werror" CXXFLAGS="-Werror" cmake -DDOWNLOAD_GTEST=ON ..
           make everything
+          make package_default
+          cp DDNet-*-linux_x86_64.tar.xz ../artifacts
```
Comment (Learath2) [inline .circleci/config.yml]: Don't rename the file to artifacts, idk if globs work but putting it in a directory and storing the directory instead definitely works.

File: .circleci/config.yml [hunk 2]
```diff
@@ -52,12 +52,17 @@ jobs:
...
       - run: |
-          mkdir build2
-          cd build2
+          mkdir noautoupdate
+          cd noautoupdate
           env CFLAGS="-Wdeclaration-after-statement -Werror" CXXFLAGS="-Werror" cmake -DAUTOUPDATE=OFF -DDOWNLOAD_GTEST=ON ..
           make everything
 
+      - store_artifacts:
+          path: artifacts
```
Comment (Learath2) [inline .circleci/config.yml]: Might want to change the destination as well, seems the path is a tad too deep.

**Comment by heinrich5991** [comment]
The artifacts are at https://circleci.com/gh/ddnet/ddnet/4632#artifacts/containers/0. You can't find the link without being logged in, but you can still access it.

**Comment by heinrich5991** [comment]
@Learath2?

**Comment by def-** [comment]
bors r+
Thanks

**Pull request merged**